### PR TITLE
Breadcrumbs for Marketing Pages e2e tests added

### PR DESCRIPTION
**Pull request by EnitanOO**
AB#15190
AB#15652

### CODE REVIEW

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/PublicBrowse/SolutionObjects.cs [hunk 1]
```diff
@@ -32,5 +32,15 @@ internal static class SolutionObjects
...
+
+        internal static By BreadcrumbsBanner => By.ClassName("nhsuk-breadcrumb__item");
+
+        internal static By CatalogueSolutionCrumb => CustomBy.DataTestId("catalogue-solutions-crumb");
+
+        internal static By HomeCrumb => CustomBy.DataTestId("home-crumb");
+
+        internal static By CatalogueSolutionPage => CustomBy.DataTestId("solutions-list-body");
+
+        internal static By HomePage => By.Id("maincontent");
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/PublicBrowse/SolutionObjects.cs:44]: HomePage check should use objects from another area. Tis is not in the right place and will only confuse matters further

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs [hunk 2]
```diff
@@ -18,6 +18,34 @@ public void ClickEpics()
...
+                Wait.Until(s => ElementDisplayed(Objects.PublicBrowse.SolutionObjects.CatalogueSolutionPage));
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public bool HomePageDisplayed()
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:35]: 2 major issues in this to me. Firstly this method is doing 2 actions (Clicking the breadcrumb and checking that the homepage is displayed), secondly the check that the homepage is displayed should not be in SolutionActions, it should be in the HomepageActions class

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs [hunk 3]
```diff
@@ -18,6 +18,34 @@ public void ClickEpics()
             Driver.FindElement(Objects.PublicBrowse.SolutionObjects.SolutionEpicLink).Click();
         }
 
+        public bool CatalogueSolutionPageDisplayed()
+        {
+            ClickCatalogueSolutionBreadcrumb();
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:23]: This should be removed from this method and added to the test instead

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs [hunk 4]
```diff
@@ -108,6 +136,28 @@ internal IEnumerable<AssociatedService> GetAssociatedServicesInfo()
...
+        }
+
+        internal void ClickCatalogueSolutionBreadcrumb()
+        {
+            Driver.FindElement(Objects.PublicBrowse.SolutionObjects.CatalogueSolutionCrumb).Click();
+        }
+
+        internal void ClickHomeBreadcrumb()
+        {
+            Driver.FindElement(Objects.PublicBrowse.SolutionObjects.CatalogueSolutionCrumb).Click();
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:158]: The wrong object is being used here

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs [hunk 5]
```diff
@@ -108,6 +136,28 @@ internal IEnumerable<AssociatedService> GetAssociatedServicesInfo()
...
 
+        internal bool BreadcrumbBannerDisplayed()
+        {
+            return ElementDisplayed(Objects.PublicBrowse.SolutionObjects.BreadcrumbsBanner);
+        }
+
+        internal string GetBreadcrumbNames(string breadcrumbItem)
+        {
+            var rows = Driver.FindElements(Objects.PublicBrowse.SolutionObjects.BreadcrumbsBanner);
+            var row = rows.Single(s => s.FindElement(By.TagName("a")).Text.Contains(breadcrumbItem, System.StringComparison.OrdinalIgnoreCase));
```
Comment (PhysicsJon) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:147]: Why have you used a lambda parameter of `s` here? would `r` not be better?

**Review by PhysicsJon — CHANGES_REQUESTED** [review]

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs (hunk 5 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:147]: done

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs (hunk 4 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:158]: corrected

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs (hunk 2 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:35]: Hopefully redone right

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs (hunk 3 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:23]: done

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs (hunk 5 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/PublicBrowse/SolutionAction.cs:147]: corrected

**Review by PhysicsJon — APPROVED** [review]

**Pull request merged**